src/Diagnostics.RuntimeHost/Services/CacheService/DiagEntityTableCacheService.cs

```diff
@@ -16,11 +16,33 @@ public class DiagEntityTableCacheService : IDiagEntityTableCacheService
 
         private IStorageService storageService;
 
+        int cacheExpirationTimeInSecs = 60;
+
         public DiagEntityTableCacheService(IStorageService service)
         {
             cache = new ConcurrentDictionary<string, List<DiagEntity>>();
             storageService = service;
+            StartPolling();
         }
+
+        private async void StartPolling()
+        {
+            do
+            {
+                await Task.Delay(cacheExpirationTimeInSecs * 1000);
+                var detectorResult = await storageService.GetEntitiesByPartitionkey("Detector");
+                if (detectorResult != null)
+                {
+                    AddOrUpdate("Detector", detectorResult);
+                }
+                var gistResult = await storageService.GetEntitiesByPartitionkey("Gist");
```

Comment:
You can potentially call these two tasks (detector task and gist task) in parallel to speed up things.